### PR TITLE
Link updates for TIKA 2262

### PR DESCRIPTION
**Pull request by ThejanW**

### CODE REVIEW

File: tika-parsers/src/main/resources/org/apache/tika/parser/captioning/tf/Im2txtRestDockerfile [hunk 1]
```diff
@@ -18,35 +18,41 @@
...
+    echo "We're downloading the checkpoint file for image captioning, the shell might look unresponsive. Please be patient."  && \
+    git clone https://github.com/USCDataScience/img2text.git && \
+    # Join the parts
+    cat img2text/models/1M_iters_ckpt_parts_a* >1M_iters_ckpt.tar.gz && \
+    tar -xzvf 1M_iters_ckpt.tar.gz && \
+    # Delete all files except 1M_iters_ckpt
+    rm -rf {1M_iters_ckpt.tar.gz,img2text}
 
+RUN \
     wget https://raw.githubusercontent.com/ThejanW/tika/master/tika-parsers/src/main/resources/org/apache/tika/parser/captioning/tf/model_info.xml \
```
Comment (ThejanW) [inline tika-parsers/src/main/resources/org/apache/tika/parser/captioning/tf/Im2txtRestDockerfile]: I've updated the model_info.xml in this PR, @chrismattmann please update this link to "https://raw.githubusercontent.com/apache/tika/master/tika-parsers/src/main/resources/org/apache/tika/parser/captioning/tf/model_info.xml"
once merged this.

**Comment by ThejanW** [comment]
@chrismattmann ping!

File: tika-parsers/src/main/resources/org/apache/tika/parser/captioning/tf/Im2txtRestDockerfile (hunk 1 above)
Comment (chrismattmann) [inline tika-parsers/src/main/resources/org/apache/tika/parser/captioning/tf/Im2txtRestDockerfile]: please update this wget line to point to apache/master, then I think we're good to test/go @ThejanW

File: tika-parsers/src/main/java/org/apache/tika/parser/captioning/tf/TensorflowRESTCaptioner.java [hunk 2]
```diff
@@ -58,9 +58,9 @@
  * Tensorflow image captioner.
  * This implementation uses Tensorflow via REST API.
  * <p>
- * NOTE : //TODO: link to wiki page here
+ * NOTE : https://wiki.apache.org/tika/ImageCaption#preview
```
Comment (chrismattmann) [inline tika-parsers/src/main/java/org/apache/tika/parser/captioning/tf/TensorflowRESTCaptioner.java]: remove the #preview part at the end

**Comment by ThejanW** [comment]
@chrismattmann Ping!

**Comment by chrismattmann** [comment]
awesome @ThejanW I'll test this today and hopefully commit!

**Comment by ThejanW** [comment]
@chrismattmann thanks. model_info.xml file is changed in this PR & in dockerfile it wget the model_info.xml from apache repo. So it won't work unless we merge the changes of model_info.xml first.

**Comment by chrismattmann** [comment]
got it @ThejanW so I'll merge first, then test and fix after that.

**Comment by ThejanW** [comment]
awesome @chrismattmann :100:

**Comment by thammegowda** [comment]
@ThejanW 
I got an error. 

```bash
docker run -p 8764:8764 -it im2txt-rest-tika
INFO:tensorflow:Building model.
CRITICAL:tensorflow:Vocab file /usr/share/apache-tika/models/dl/image/caption/model_main/word_counts.txt not found.
INFO:tensorflow:Initializing vocabulary from file: /usr/share/apache-tika/models/dl/image/caption/model_main/word_counts.txt
Traceback (most recent call last):
  File "/usr/bin/im2txtapi", line 129, in <module>
    app = Initializer(__name__)
  File "/usr/bin/im2txtapi", line 114, in __init__
    self.vocab = vocabulary.Vocabulary(FLAGS.vocab_file)
  File "/usr/share/apache-tika/src/dl/image/caption/vocabulary.py", line 41, in __init__
    reverse_vocab = list(f.readlines())
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/lib/io/file_io.py", line 128, in readlines
    self._preread_check()
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/lib/io/file_io.py", line 73, in _preread_check
    compat.as_bytes(self.__name), 1024 * 512, status)
  File "/usr/lib/python2.7/contextlib.py", line 24, in __exit__
    self.gen.next()
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/framework/errors_impl.py", line 466, in raise_exception_on_not_ok_status
    pywrap_tensorflow.TF_GetCode(status))
tensorflow.python.framework.errors_impl.NotFoundError: /usr/share/apache-tika/models/dl/image/caption/model_main/word_counts.txt

```
Here is how I tested it  (complete log) (let me know I missed a step or two):
 
```bash
✔ ~/work/projects/apache/tika/tika-parsers/src/main/resources/org/apache/tika/parser/captioning/tf [ThejanW-master L|✔] 
11:40 $ docker build . -f Im2txtRestDockerfile  -t im2txt-rest-tika
Sending build context to Docker daemon 40.45 kB
Step 1/12 : FROM inception-rest-tika
repository inception-rest-tika not found: does not exist or no pull access
✘-1 ~/work/projects/apache/tika/tika-parsers/src/main/resources/org/apache/tika/parser/captioning/tf [ThejanW-master L|✔] 
11:40 $ cd -
/home/tg/work/projects/apache/tika/tika-parsers/src/main/resources/org/apache/tika/parser/recognition/tf
✔ ~/work/projects/apache/tika/tika-parsers/src/main/resources/org/apache/tika/parser/recognition/tf [ThejanW-master L|✔] 
11:40 $ ls
classify_image.py  inceptionapi.py  InceptionRestDockerfile  InceptionVideoRestDockerfile  video_util.py
✔ ~/work/projects/apache/tika/tika-parsers/src/main/resources/org/apache/tika/parser/recognition/tf [ThejanW-master L|✔] 
11:40 $ docker build . -f InceptionRestDockerfile  -t inception-rest-tika 
Sending build context to Docker daemon 41.98 kB
Step 1/11 : FROM ubuntu
 ---> f7b3f317ec73
Step 2/11 : MAINTAINER Thamme Gowda <thammegowda@apache.org> and Kranthi Kiran GV <kkranthi@student.nitw.ac.in>
 ---> Using cache
 ---> 223d090ede88
Step 3/11 : RUN apt-get update &&     apt-get install -y python-pip python-dev wget libtcmalloc-minimal4 git unzip
 ---> Using cache
 ---> 926c3b6278fa
Step 4/11 : RUN pip install --upgrade https://storage.googleapis.com/tensorflow/linux/cpu/tensorflow-1.0.1-cp27-none-linux_x86_64.whl &&   pip install flask requests
 ---> Using cache
 ---> 6691de0748ec
Step 5/11 : RUN wget https://github.com/tensorflow/models/archive/c15fada28113eca32dc98d6e3bec4755d0d5b4c2.zip && 	unzip c15fada28113eca32dc98d6e3bec4755d0d5b4c2.zip
 ---> Using cache
 ---> cf489cd40efe
Step 6/11 : RUN wget https://raw.githubusercontent.com/apache/tika/master/tika-parsers/src/main/resources/org/apache/tika/parser/recognition/tf/inceptionapi.py -O /usr/bin/inceptionapi.py &&   chmod +x /usr/bin/inceptionapi.py
 ---> Using cache
 ---> eef1348f81e3
Step 7/11 : RUN apt-get clean
 ---> Using cache
 ---> 7df53942f3f1
Step 8/11 : ENV PYTHONPATH "$PYTHONPATH:/models-c15fada28113eca32dc98d6e3bec4755d0d5b4c2/slim"
 ---> Using cache
 ---> 5b9e8530f435
Step 9/11 : ENV LD_PRELOAD "/usr/lib/libtcmalloc_minimal.so.4"
 ---> Using cache
 ---> a49f1889b847
Step 10/11 : EXPOSE 8764
 ---> Using cache
 ---> ee023af6eaa4
Step 11/11 : CMD inceptionapi.py
 ---> Using cache
 ---> 39f1b6a628ed
Successfully built 39f1b6a628ed
✔ ~/work/projects/apache/tika/tika-parsers/src/main/resources/org/apache/tika/parser/recognition/tf [ThejanW-master L|✔] 
11:41 $ cd -
/home/tg/work/projects/apache/tika/tika-parsers/src/main/resources/org/apache/tika/parser/captioning/tf
✔ ~/work/projects/apache/tika/tika-parsers/src/main/resources/org/apache/tika/parser/captioning/tf [ThejanW-master L|✔] 
11:41 $ docker build . -f Im2txtRestDockerfile  -t im2txt-rest-tika
Sending build context to Docker daemon 40.45 kB
Step 1/12 : FROM inception-rest-tika
 ---> 39f1b6a628ed
Step 2/12 : MAINTAINER Apache Tika Team
 ---> Using cache
 ---> 39f156f76d63
Step 3/12 : RUN pip install pillow
 ---> Using cache
 ---> 2da7825cccdf
Step 4/12 : WORKDIR /usr/share/apache-tika/models/dl/image/caption/
 ---> Using cache
 ---> 8fbd9d6cf6b0
Step 5/12 : RUN git config --global http.postBuffer 1048576000 &&     echo "We're downloading the checkpoint file for image captioning, the shell might look unresponsive. Please be patient."  &&     git clone -b models https://github.com/USCDataScience/img2text.git &&     cat img2text/models/1M_iters_ckpt_parts_a* >1M_iters_ckpt.tar.gz &&     tar -xzvf 1M_iters_ckpt.tar.gz &&     rm -rf {1M_iters_ckpt.tar.gz,img2text}
 ---> Running in 882e5314413f
We're downloading the checkpoint file for image captioning, the shell might look unresponsive. Please be patient.
Cloning into 'img2text'...
1M_iters_ckpt/model.ckpt-1000000.data-00000-of-00001
1M_iters_ckpt/model.ckpt-1000000.index
1M_iters_ckpt/word_counts.txt
1M_iters_ckpt/
 ---> caa596bcc989
Removing intermediate container 882e5314413f
Step 6/12 : RUN wget https://raw.githubusercontent.com/apache/tika/master/tika-parsers/src/main/resources/org/apache/tika/parser/captioning/tf/model_info.xml         -O model_info.xml
 ---> Running in fafa53202d74
--2017-07-22 18:42:22--  https://raw.githubusercontent.com/apache/tika/master/tika-parsers/src/main/resources/org/apache/tika/parser/captioning/tf/model_info.xml
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 151.101.0.133, 151.101.64.133, 151.101.128.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|151.101.0.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 1129 (1.1K) [text/plain]
Saving to: 'model_info.xml'

     0K .                                                     100% 32.2M=0s

2017-07-22 18:42:22 (32.2 MB/s) - 'model_info.xml' saved [1129/1129]

 ---> 6e1f9b14e031
Removing intermediate container fafa53202d74
Step 7/12 : WORKDIR /usr/share/apache-tika/src/dl/image/caption/
 ---> 0ffc054e6bc3
Removing intermediate container d3b1a1c284bd
Step 8/12 : RUN wget https://raw.githubusercontent.com/apache/tika/master/tika-parsers/src/main/resources/org/apache/tika/parser/captioning/tf/caption_generator.py         -O caption_generator.py &&     wget https://raw.githubusercontent.com/apache/tika/master/tika-parsers/src/main/resources/org/apache/tika/parser/captioning/tf/im2txtapi.py         -O im2txtapi.py &&     wget https://raw.githubusercontent.com/apache/tika/master/tika-parsers/src/main/resources/org/apache/tika/parser/captioning/tf/model_wrapper.py         -O model_wrapper.py &&     wget https://raw.githubusercontent.com/apache/tika/master/tika-parsers/src/main/resources/org/apache/tika/parser/captioning/tf/vocabulary.py         -O vocabulary.py &&     chmod +x im2txtapi.py
 ---> Running in 83c88039497d
--2017-07-22 18:42:23--  https://raw.githubusercontent.com/apache/tika/master/tika-parsers/src/main/resources/org/apache/tika/parser/captioning/tf/caption_generator.py
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 151.101.0.133, 151.101.64.133, 151.101.128.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|151.101.0.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 6320 (6.2K) [text/plain]
Saving to: 'caption_generator.py'

     0K ......                                                100% 9.81M=0.001s

2017-07-22 18:42:23 (9.81 MB/s) - 'caption_generator.py' saved [6320/6320]

--2017-07-22 18:42:23--  https://raw.githubusercontent.com/apache/tika/master/tika-parsers/src/main/resources/org/apache/tika/parser/captioning/tf/im2txtapi.py
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 151.101.0.133, 151.101.64.133, 151.101.128.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|151.101.0.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 8981 (8.8K) [text/plain]
Saving to: 'im2txtapi.py'

     0K ........                                              100% 2.57M=0.003s

2017-07-22 18:42:23 (2.57 MB/s) - 'im2txtapi.py' saved [8981/8981]

--2017-07-22 18:42:23--  https://raw.githubusercontent.com/apache/tika/master/tika-parsers/src/main/resources/org/apache/tika/parser/captioning/tf/model_wrapper.py
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 151.101.0.133, 151.101.64.133, 151.101.128.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|151.101.0.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 13099 (13K) [text/plain]
Saving to: 'model_wrapper.py'

     0K .......... ..                                         100% 2.44M=0.005s

2017-07-22 18:42:23 (2.44 MB/s) - 'model_wrapper.py' saved [13099/13099]

--2017-07-22 18:42:23--  https://raw.githubusercontent.com/apache/tika/master/tika-parsers/src/main/resources/org/apache/tika/parser/captioning/tf/vocabulary.py
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 151.101.0.133, 151.101.64.133, 151.101.128.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|151.101.0.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 2464 (2.4K) [text/plain]
Saving to: 'vocabulary.py'

     0K ..                                                    100% 9.47M=0s

2017-07-22 18:42:24 (9.47 MB/s) - 'vocabulary.py' saved [2464/2464]

 ---> a8c11e079b30
Removing intermediate container 83c88039497d
Step 9/12 : WORKDIR /
 ---> 5c3d8c3c2959
Removing intermediate container 6846ff498d1e
Step 10/12 : RUN ln -s /usr/share/apache-tika/src/dl/image/caption/im2txtapi.py /usr/bin/im2txtapi
 ---> Running in 2a0784c5ca1f
 ---> f4e7c1b74baf
Removing intermediate container 2a0784c5ca1f
Step 11/12 : EXPOSE 8764
 ---> Running in d59c226df677
 ---> bbd3abf0e8cf
Removing intermediate container d59c226df677
Step 12/12 : CMD im2txtapi
 ---> Running in 838f665704f9
 ---> ae97eef0e39c
Removing intermediate container 838f665704f9
Successfully built ae97eef0e39c
✔ ~/work/projects/apache/tika/tika-parsers/src/main/resources/org/apache/tika/parser/captioning/tf [ThejanW-master L|✔] 
11:42 $ docker run -p 8764:8764 -it im2txt-rest-tika
INFO:tensorflow:Building model.
CRITICAL:tensorflow:Vocab file /usr/share/apache-tika/models/dl/image/caption/model_main/word_counts.txt not found.
INFO:tensorflow:Initializing vocabulary from file: /usr/share/apache-tika/models/dl/image/caption/model_main/word_counts.txt
Traceback (most recent call last):
  File "/usr/bin/im2txtapi", line 129, in <module>
    app = Initializer(__name__)
  File "/usr/bin/im2txtapi", line 114, in __init__
    self.vocab = vocabulary.Vocabulary(FLAGS.vocab_file)
  File "/usr/share/apache-tika/src/dl/image/caption/vocabulary.py", line 41, in __init__
    reverse_vocab = list(f.readlines())
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/lib/io/file_io.py", line 128, in readlines
    self._preread_check()
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/lib/io/file_io.py", line 73, in _preread_check
    compat.as_bytes(self.__name), 1024 * 512, status)
  File "/usr/lib/python2.7/contextlib.py", line 24, in __exit__
    self.gen.next()
  File "/usr/local/lib/python2.7/dist-packages/tensorflow/python/framework/errors_impl.py", line 466, in raise_exception_on_not_ok_status
    pywrap_tensorflow.TF_GetCode(status))
tensorflow.python.framework.errors_impl.NotFoundError: /usr/share/apache-tika/models/dl/image/caption/model_main/word_counts.txt
✘-1 ~/work/projects/apache/tika/tika-parsers/src/main/resources/org/apache/tika/parser/captioning/tf [ThejanW-master L|✔] 
✘-1 ~/work/projects/apache/tika/tika-parsers/src/main/resources/org/apache/tika/parser/captioning/tf [ThejanW-master L|✔] 
11:42 $ docker --version
Docker version 17.03.1-ce, build c6d412e
✔ ~/work/projects/apache/tika/tika-parsers/src/main/resources/org/apache/tika/parser/captioning/tf [ThejanW-master L|✔] 
11:42 $ lsb_release -a
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 17.04
Release:	17.04
Codename:	zesty

```

**Comment by ThejanW** [comment]
yes..you are getting an error because you have not pushed the model_info.xml file into Apache repo. I have edited the model_info.xml file in this PR & it has updated directories. Please merge model_info.xml file first, rebuild container and test.

**Comment by ThejanW** [comment]
@thammegowda please see the conversation I had with @chrismattmann.

**Comment by thammegowda** [comment]
> Please merge model_info.xml file first, rebuild container and test.

It's complicated!  It's a deadlock - We can't merge until tests pass, and (by the way links are pointing in docker file) tests won't pass until it gets merged. 
For future scenarios like this, please split the changes into two PR's. Updates to `model_info.xml` should be merged first.

Two options I have now:

1. Raise a new PR with the changes to model_info file, merge it first. then it will create conflicts with the current PR. Then ask you to fix it.
OR
2. I take a leap of faith, copy changes you have made to `model_info.xml` to the master branch after basic testing. 

Going with the option 2, (I tested by editing a link to the incoming file in PR, it worked.) I am going to merge this.

**Comment by chrismattmann** [comment]
I've also confirmed this worked:

```
 ---> 882bb8b17b6f
Removing intermediate container 6536ff83fcca
Successfully built 882bb8b17b6f
LMC-053601:tf mattmann$ docker run -it -p 8764:8764 im2txt-rest-tika
INFO:tensorflow:Building model.
INFO:tensorflow:Initializing vocabulary from file: /usr/share/apache-tika/models/dl/image/caption/1M_iters_ckpt/word_counts.txt
INFO:tensorflow:Created vocabulary with 11520 words
W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use SSE3 instructions, but these are available on your machine and could speed up CPU computations.
W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use SSE4.1 instructions, but these are available on your machine and could speed up CPU computations.
W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use SSE4.2 instructions, but these are available on your machine and could speed up CPU computations.
W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use AVX instructions, but these are available on your machine and could speed up CPU computations.
INFO:tensorflow:Loading model from checkpoint: /usr/share/apache-tika/models/dl/image/caption/1M_iters_ckpt/model.ckpt-1000000
INFO:tensorflow:Successfully loaded checkpoint: model.ckpt-1000000
Serving on port 8764
 * Running on http://0.0.0.0:8764/ (Press CTRL+C to quit)
10.0.2.2 - - [09/Aug/2017 15:40:21] "GET /inception/v3/captions?url=http://img09.deviantart.net/5281/i/2009/326/a/f/girl_walk_in_the_water_by_jmd4040116.jpg&beam_size=3&max_caption_length=15 HTTP/1.1" 200 -
10.0.2.2 - - [09/Aug/2017 15:40:21] "GET /favicon.ico HTTP/1.1" 404 -
```

Great job.

**Pull request merged**